docs/DeveloperGuide.md

```diff
@@ -1,33 +1,297 @@
+    - "add-sleep hours/8 datetime/2021-09-01 06:00" for adding sleep.
+    - "edit-sleep 1 hours/8 datetime/2021-09-01 06:00" for editing sleep.
+    - "delete-sleep 1" for deleting sleep.
+    - "list-sleep" for listing all sleep.
+
+2. **Command Identification**: The Parser Class identifies the type of sleep operation and passes the necessary parameters.
+
+3. **Command Creation**: An instance of the corresponding command class is created (e.g., AddSleepCommand, EditSleepCommand, etc.) and returned to AthletiCLI.
+
+4. **Command Execution**: AthletiCLI executes the command, interacting with the data instance of SleepList to perform the required operation.
+
+5. **Result Display**: A message is returned post-execution and passed through AthletiCLI to the UI for display to the user.
+
+The following class diagram shows how sleep and sleep-related classes are constructed in AthletiCLI:
+
+<p  align="center" width="100%">
+  <img width="80%" src="images/SleepAndSleepListClassDiagram.svg" alt="Class Diagram of Sleep and SleepList"/>
+
+</p>
 
 
 ## Product scope
 ### Target user profile
 
-{Describe the target user profile}
+AthletiCLI is designed for athletic individuals who are committed to optimizing their performance. 
+
+These users are highly disciplined and engaged not only in regular, intense physical training but also in nutrition, mental conditioning, and recovery. 
+
+They are looking for a holistic tool that integrates all facets of an athletic lifestyle. AthletiCLI serves as a daily or weekly companion, designed to monitor, track, and analyze various elements crucial for high-level athletic performance. 
 
 ### Value proposition
 
-{Describe the value proposition: what problem does it solve?}
+AthletiCLI provides a streamlined, integrated solution for athletic individuals focused on achieving peak performance. 
+
+While the app includes robust capabilities for tracking physical training metrics, it also offers features for monitoring dietary habits and sleep metrics. 
+
+By providing a comprehensive view of various performance-related factors over time, AthletiCLI enables athletes to identify trends, refine their training and lifestyle habits, and optimize outcomes. The app is more than a tracking tool—it's a performance optimization platform that takes into account the full spectrum of an athlete's life. 
 
 ## User Stories
 
-|Version| As a ... | I want to ... | So that I can ...|
-|--------|----------|---------------|------------------|
-|v1.0|new user|see usage instructions|refer to them when I forget how to use the application|
-|v2.0|user|find a to-do item by name|locate a to-do without having to go through the entire list|
+| Version | As a ...                        | I want to ...                                                     | So that I can ...                                                                      |
```

Comment:
<img width="802" alt="image" src="https://github.com/nus-cs2113-AY2324S1/tp/assets/34310305/94ee9f02-11c4-4488-a0ba-42a7485778ea">
Perhaps a space between 'Version' and 'As a...' would make things more readable.